src/Collection/Iterator/UnfoldIterator.php

```diff
@@ -57,34 +47,76 @@ class UnfoldIterator extends IteratorIterator implements RecursiveIterator
      */
     public function __construct(Traversable $items, callable $unfolder)
     {
-        $this->_unfolder = $unfolder;
-        parent::__construct($items);
-        $this->_innerIterator = $this->getInnerIterator();
+        $itemsIterator = new class ($items, $unfolder) extends IteratorIterator implements RecursiveIterator {
```

Comment:
Instead of unfold() using a RecursiveIteratorIterator, UnfoldIterator wraps the items itself so it can return a custom key() for each result.